htlcswitch/switch.go

```diff
@@ -493,14 +499,11 @@ func (s *Switch) GetAttemptResult(attemptID uint64, paymentHash lntypes.Hash,
 	// Since the attempt was known, we can start a goroutine that can
 	// extract the result when it is available, and pass it on to the
 	// caller.
-	s.wg.Add(1)
-	go func() {
-		defer s.wg.Done()
-
+	ok := s.gm.Go(background, func(ctx context.Context) {
 		var n *networkResult
 		select {
 		case n = <-nChan:
-		case <-s.quit:
+		case <-s.gm.Done():
```

Comment:
i think it is not great to refer to `s.gm` from inside a call-back that is called from `s.gm`. Rather just use the `ctx` provided to the callback which will be cancelled when the gm is shutdown (ie, when gm.Done() would have returned anyways)